### PR TITLE
Add TCLAP v1.2.3

### PR DESCRIPTION
**Pull request by xguerin**
Specify library name and version:  **tclap/1.2.3**

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/tclap/1.2.3/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+from conans import ConanFile, tools
+
+class TclapConan(ConanFile):
+    name = "tclap"
+    version = "1.2.3"
+    license = "MIT License"
```
Comment (prince-chrismc) [inline recipes/tclap/1.2.3/conanfile.py:6]: Please you the SPDX short identifiers

```suggestion
    license = "MIT"
```

File: recipes/tclap/1.2.3/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+from conans import ConanFile, tools
+
+class TclapConan(ConanFile):
+    name = "tclap"
+    version = "1.2.3"
```
Comment (prince-chrismc) [inline recipes/tclap/1.2.3/conanfile.py:5]:
```suggestion
```

You can set the version at the command line `conan create conanfile.py 1.2.3@`

File: recipes/tclap/1.2.3/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    license = "MIT License"
+    homepage = "http://github.com/xguerin/tclap"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "Templatized Command Line Argument Parser"
+    topics = ("c++", "commandline parser")
+    no_copy_source = True
+    exports_sources = ["include/*"]
```
Comment (prince-chrismc) [inline recipes/tclap/1.2.3/conanfile.py:12]:
```suggestion

    @property
    def _source_subfolder(self):
        return "source_subfolder"
```

File: recipes/tclap/1.2.3/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
```
Comment (prince-chrismc) [inline recipes/tclap/1.2.3/conanfile.py:15]:
```suggestion
        tools.get(**self.conan_data["sources"][self.version])
         os.rename(self.name + "-" + self.version, self._source_subfolder)
```

File: recipes/tclap/1.2.3/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+    def package(self):
+        src_folder = self.name + "-" + self.version
+        self.copy("*.h", "include", src_folder + "/include")
```
Comment (prince-chrismc) [inline recipes/tclap/1.2.3/conanfile.py:19]: https://github.com/xguerin/tclap/blob/6689a6109700390d61c44298181da7e73fab7dae/tclap.pc.in#L4

```suggestion
        self.copy("COPYING", src=self._source_subfolder, dst="licenses")
        self.copy(pattern="*", src=os.path.join(self._source_subfolder, "include"), dst="include", keep_path=True)

    def package_info(self):
        self.cpp_info.names["pkg_config"] = "tclap"
```

File: recipes/tclap/1.2.3/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    homepage = "http://github.com/xguerin/tclap"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "Templatized Command Line Argument Parser"
+    topics = ("c++", "commandline parser")
+    no_copy_source = True
+    exports_sources = ["include/*"]
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+
```
Comment (prince-chrismc) [inline recipes/tclap/1.2.3/conanfile.py:16]:
```suggestion

    def package_id(self):
        self.info.header_only()

```

File: recipes/tclap/1.2.3/test_package/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+        del self.settings.compiler.libcxx
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def test(self):
+        if "x86" in self.settings.arch and not tools.cross_building(self.settings):
+            self.run(os.path.join("bin", "test"), run_environment=True)
```
Comment (prince-chrismc) [inline recipes/tclap/1.2.3/test_package/conanfile.py:18]:
```suggestion
        if not tools.cross_building(self.settings):
            bin_path = os.path.join("bin", "test_package")
            self.run(bin_path, run_environment=True)
```

File: recipes/tclap/1.2.3/test_package/conanfile.py [hunk 8]
```diff
@@ -0,0 +1,18 @@
+import os
+from conans import ConanFile, CMake, tools
+
+class TestTclapConan(ConanFile):
+    settings = "os", "compiler", "arch", "build_type"
+    generators = "cmake"
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+
```
Comment (prince-chrismc) [inline recipes/tclap/1.2.3/test_package/conanfile.py:10]:
```suggestion
```

File: recipes/tclap/1.2.3/test_package/CMakeLists.txt [hunk 9]
```diff
@@ -0,0 +1,9 @@
+cmake_minimum_required(VERSION 3.0)
+project(test_package CXX)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup(TARGETS)
+
+add_executable(test_tclap test.cpp)
+target_link_libraries(test_tclap CONAN_PKG::tclap)
+set_target_properties(test_tclap PROPERTIES OUTPUT_NAME "test")
```
Comment (prince-chrismc) [inline recipes/tclap/1.2.3/test_package/CMakeLists.txt:9]:
```suggestion
add_executable(test_package test.cpp)
target_link_libraries(test_package CONAN_PKG::tclap)
```

**Review by prince-chrismc — CHANGES_REQUESTED** [review]
Along with all the comments... If you could please take a look at the docs https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md#the-version-folders

Please rename the folder from `1.2.3` to `all` since this project is very unlikely to change over time

**Comment by prince-chrismc** [comment]
Please try to avoid force pushing it makes reviewing much more difficult

**Review by prince-chrismc — APPROVED** [review]

File: recipes/tclap/all/test_package/CMakeLists.txt [hunk 10]
```diff
@@ -0,0 +1,8 @@
+cmake_minimum_required(VERSION 3.0)
```
Comment (theirix) [inline recipes/tclap/all/test_package/CMakeLists.txt:1]:
```suggestion
cmake_minimum_required(VERSION 3.1)
```

**Review by theirix — CHANGES_REQUESTED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**